web3/eth.py

```diff
@@ -749,11 +782,43 @@ def sign_munger(
         mungers=[default_root_munger],
     )
 
-    call: Method[Callable[..., Union[bytes, bytearray]]] = Method(
+    _call: Method[Callable[..., Union[bytes, bytearray]]] = Method(
         RPC.eth_call,
         mungers=[BaseEth.call_munger]
     )
 
+    def call(
+        self,
+        transaction: TxParams,
+        block_identifier: Optional[BlockIdentifier] = None,
+        state_override: Optional[CallOverride] = None,
+    ) -> Union[bytes, bytearray]:
+        if self.w3.provider.ccip_read_enabled:
+            # if ccip_read_enabled, handle OffchainLookup reverts appropriately
+            self.durin_call(transaction, block_identifier, state_override)
+
+        return self._call(transaction, block_identifier, state_override)
+
+    def durin_call(
+        self,
+        transaction: TxParams,
+        block_identifier: Optional[BlockIdentifier] = None,
+        state_override: Optional[CallOverride] = None,
+    ) -> Union[bytes, bytearray]:
+        max_redirects = self.w3.provider.ccip_read_max_redirects
+
+        if not max_redirects or max_redirects < 4:
+            raise ValueError("ccip_read_max_redirects property on provider must be at least 4.")
+
+        for _ in range(max_redirects):
+            try:
+                return self._call(transaction, block_identifier, state_override)
+            except OffchainLookup as offchain_lookup:
+                durin_calldata = handle_offchain_lookup(offchain_lookup.payload, transaction)
+                transaction['data'] = durin_calldata
```

Comment:
I assume we don't, but just to be sure, do we need to return something here?

Reply:
Yeah we don't. All we're doing is replacing the `data` in the transaction which was built from the offchain REST request and the `OffchainLookup` information and calling that in the `try` block again.